BBC.js

```diff
@@ -1,148 +1,217 @@
-		 if(temp)
-		 { 	newItem.abstractNote=temp.value;     }
-		 else
-		 {
-			 xpath='//meta[@name="description"]/@content';
-		 		 temp=doc.evaluate(xpath, doc, nsResolver,XPathResult.ANY_TYPE, null).iterateNext();
-				 if(temp)
-				 { 	newItem.abstractNote=temp.value;     }
-		 }
-		 
-		 newItem.attachments.push({url:url, title:"BBC News Snapshot",mimeType:"text/html"});
-		 
-		 newItem.complete();
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//a[h3[@class="title-link__title"]]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
 }
 
 
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
 
-function doWeb(doc,url)
-{
-	   var namespace = doc.documentElement.namespaceURI;
-	  var nsResolver = namespace ? function(prefix) {
-	  if (prefix == 'x') return namespace; else return null;
-	  } : null;
-	  
-	  var xpath='//meta[@name="Headline"]/@content';
-	  var title;
-	 if(title=doc.evaluate(xpath, doc, nsResolver,XPathResult.ANY_TYPE, null).iterateNext())
-	 	{  scrape(doc,url,title.value) }
-	 else
-	 {
-		 xpath='//font[@class="poshead"]/b';
-		 if(title=doc.evaluate(xpath, doc, nsResolver,XPathResult.ANY_TYPE, null).iterateNext())
-			 	{   scrape(doc,url,title.textContent)  }
-	 }
+function scrape(doc, url) {
+	
+	var itemType = detectWeb(doc, url);
+	
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	translator.setDocument(doc);
+	
+	translator.setHandler('itemDone', function (obj, item) {
+		
+		//add date and time if missing by one of three attempts:
+		// 1. look at the json-ld data
+		// 2. calculate it from the data-seconds attribute
+		// 3. extract it from a nonstandard meta field
+		var jsonld = ZU.xpathText(doc, '//script[@type="application/ld+json"]');
+		var data = JSON.parse(jsonld);
+		//Z.debug(data);
+		if (data && data.datePublished) {
+			item.date = data.datePublished;
+		} else {
+			var seconds = ZU.xpathText(doc, '(//div[h1 or h2]//*[contains(@class, "date")]/@data-seconds)[1]');
+			if (!item.date && seconds) {
+				//Z.debug(seconds);
+				var date = new Date(1000*seconds);
+				item.date = date.toISOString();
+			} else {
+				item.date = ZU.xpathText(doc, '//meta[@property="rnews:datePublished"]/@content');
+			}
+		}
+		
+		//delete wrongly attached creators like
+		//"firstName": "B. B. C.", "lastName": "News"
+		item.creators = [];
```

Comment:
But we should still try to grab creators. From http://www.bbc.com/news/magazine-15335899
it looks like `//span[@class="byline__name"]` should work